src/model/tests/JointUnitTest.cpp

```diff
@@ -236,6 +237,28 @@ VectorDynSize setRandomJointVelocity(const JointType& joint, VectorDynSize& join
     return jointPos;
 }
 
+template<typename JointType>
+JointType getRandomJoint()
+{
+    // Create a random joint connecting links 0 and 1
+    JointType joint(0, 1, getRandomTransform(), getRandomAxis());
+    joint.setPosCoordsOffset(0);
+    joint.setDOFsOffset(0);
+    return joint;
+}
+
+// Specialization for SphericalJoint
+template<>
+SphericalJoint getRandomJoint<SphericalJoint>()
+{
+    SphericalJoint joint;
+    // Attach links and set a rotation-only rest transform (origins must coincide)
+    joint.setAttachedLinks(0, 1);
+    joint.setRestTransform(getRandomTransform());
+    joint.setPosCoordsOffset(0);
```

Comment:
@Nicogene shouldn't we also randomize the center here, as done for revolute/prismatic joints with `getRandomAxis` ? Otherwise we only test that weird "center in the middle of the frames" default.